.github/workflows/nightly-release.yml

```diff
@@ -0,0 +1,75 @@
+name: Nightly Release
```

Comment:
the name should be changed to something more meaningful.
`Nightly Release` means a release every night.
Maybe `pre-release` or something like that.
Each merge on main could be considered a pre-release.